app/src/main/java/com/zulip/android/networking/AsyncGetEvents.java

```diff
@@ -155,6 +161,7 @@ public void run() {
                 } catch (JSONException e) {
                     backoff(e);
                 }
```

Comment:
What we do in the web UI is this sort of exponential backoff if the client receives an error from the server (and no backoff other than the fact that the client uses longpolling (aka doesn't pass `dont_block=True` and thus is rate-limited by new events showing up at the server if it receives a success response).  Here's the code for the exponential backoff size:  https://github.com/zulip/zulip/blob/master/static/js/server_events.js#L353

Note that in this case we also show a user-facing error message overlay over the website that explains the client can't connect to the server and with a "try now" button to avoid the user having to wait up to 90s in the event that they've been off the network for a while and just got back on.  

You can see how this looks by logging into a Zulip development server and then running ctrl-C on run-dev.py and watching.

(Since this is a complicated explanation, let me know if you have any questions!)